R/caption.R

```diff
@@ -0,0 +1,110 @@
+#' Goal is to preserve space in between short and main caption
+#' when short is repeated. If short not repeated, then ok to just trim.
+#' trimws() is called quite a bit, but important to consider context of the
+#' call.
+#'
+#' @param text a chunk of caption text that may or may not need to be parsed.
+#' @param short optional short caption text
+#' @noRd
+parse_caption <- function(text, short = NULL, short_repeat = TRUE, short_sep = "") {
+  text <- trimws(text)
+  parsed_short <- str_extract(text, "^\\[.+?\\]")
+  has_parsed_short <- !is.na(parsed_short)
+  if(is.null(short) && !is.na(parsed_short)) {
```

Comment:
The previous line already put the second part of this in a variable, so `is.null(short) && !is.na(parsed_short)` → `is.null(short) && has_parsed_short`?

Reply:
Good point. This was refactored out when I more carefully pull the short title out.